router/core/factoryresolver.go

```diff
@@ -369,6 +379,15 @@ func (l *Loader) Load(engineConfig *nodev1.EngineConfiguration, subgraphs []*nod
 			}
 
 			dataSourceName := l.subgraphName(subgraphs, in.Id)
+
+			factory, err := l.resolver.ResolveGraphqlFactory(dataSourceName)
+			if err != nil {
+				return nil, err
+			}
+			if factory == nil {
```

Comment:
When can this happen?

Reply:
I can look into this, but I just moved this code to make the subgraph name available, the error handling here is as it was.

Reply:
I checked, and this appears to be logically impossible, but it is pre-existing in all the factory consumers in this file. I'll ping the git blame before removing them.

Reply:
Removed

Reply:
My fault. I wrote it like this to be extra safe but we should just trust the contract (and that the contract will never change).